### PR TITLE
Added a "Getting Involved" table.

### PR DESCRIPTION
**Pull request by dynamicwebpaige**
## Why are these changes needed?

Converting the existing "getting involved" section into a table, along with a few additional columns to set expectations. Discussion appreciated!

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -392,14 +392,40 @@ More Information
...
+   :widths: 25 50 25 25
+   :header-rows: 1
+
+   * - Platform
+     - Purpose
+     - Estimated Response Time
+     - Support Level
+   * - `Discourse Forum`_
+     - For discussions about development and questions about usage.
+     - < 2 day
```
Comment (richardliaw) [inline README.rst:405]: How about let's make this <1 day? 

Generally we're pretty good at this... and redirecting more traffic to Discourse will then further validate investment in this area!

Comment (dynamicwebpaige) [reply README.rst:405]: SGTM! Making the change now to `< 1 day` for Discourse forum responses. 😄

**Review by richardliaw — APPROVED** [review]
thanks a bunch!!!

**Comment by pcmoritz** [comment]
Congrats on your first PR!!!

**Comment by dynamicwebpaige** [comment]
✨🤩✨ *framing, putting on wall*

**Pull request merged**